src/codegen/compile.ml

```diff
@@ -8979,7 +8880,7 @@ module FuncDec = struct
     if Type.is_shared_sort sort
     then begin
       let (fi, fill) = E.reserve_fun pre_env name in
-      ( Const.Message fi, fun env ae ->
+      ( Const.Message (Wasm.I64_convert.extend_i32_s fi), fun env ae ->
```

Comment:
Hmm, if the Const.Messageargument is a function table index maybe it should remain 32-bit (I think all table indices are 32-bit, even with wasm64)

Reply:
I agree. This can stay 32-bit. (My porting was too eager...)

Reply:
**Update**: I adjusted it back to `i32`.